src/main.rs

```diff
@@ -611,21 +629,34 @@ const APP: () = {
 
         tim2_setup(&dp.TIM2);
 
+        let i2c2 = dp.I2C2;
+        i2c::setup(&rcc, &i2c2);
+
         eth::setup(&rcc, &dp.SYSCFG);
         eth::setup_pins(&dp.GPIOA, &dp.GPIOB, &dp.GPIOC, &dp.GPIOG);
 
         // c.schedule.tick(Instant::now()).unwrap();
 
         init::LateResources {
             spi: (spi1, spi2, spi4, spi5),
+            i2c: i2c2,
             ethernet_periph: (dp.ETHERNET_MAC, dp.ETHERNET_DMA, dp.ETHERNET_MTL),
         }
     }
 
-    #[idle(resources = [ethernet, ethernet_periph, iir_state, iir_ch])]
+    #[idle(resources = [ethernet, ethernet_periph, iir_state, iir_ch, i2c])]
     fn idle(c: idle::Context) -> ! {
         let (MAC, DMA, MTL) = c.resources.ethernet_periph;
 
+        let hardware_addr = match eeprom::read_eui48(c.resources.i2c) {
+            Err(_) => {
+                info!("Could not read EEPROM, using default MAC address");
+                net::wire::EthernetAddress([0x10, 0xE2, 0xD5, 0x00, 0x03, 0x00])
+            },
+            Ok(raw_mac) => net::wire::EthernetAddress(raw_mac)
+        };
+        info!("MAC: {}", hardware_addr);
+
         let hardware_addr = net::wire::EthernetAddress([0x10, 0xE2, 0xD5, 0x00, 0x03, 0x00]);
```

Comment:
I guess you didn't test this because you'd have to hack your hardware. But with this line it won't work.

Reply:
Oops - I did test this on my hardware, but I must have screwed up the final rebase.